app/conf/config.go

```diff
@@ -79,8 +83,144 @@ func LoadYamlConfig(filename string, env string) []string {
 		app.CurrentEnvironment.Port = "4100"
 	}
 
-	app.SyncQueues.Lock()
+	err = createSqsQueues(env)
+	if err != nil {
+		return ports
+	}
+
+	err = createSNSTopics(env)
+	if err != nil {
+		return ports
+	}
+
+	return ports
+}
+
+func StartWatcher(filename string, env string) {
+	quit := make(chan struct{})
+	//create watcher
+	if filename == "" {
+		filename, _ = filepath.Abs("./app/conf/goaws.yaml")
+	}
+	log.Infof("Starting watcher on file: %v", filename)
+
+	watcher, err := fsnotify.NewWatcher()
+	defer watcher.Close()
+
+	if err != nil {
+		log.Errorf("err: %s", err)
+	}
+
+	// Start listening for events.
+	go func() {
+		for {
+			select {
+			case event, ok := <-watcher.Events:
+				if !ok {
+					return
+				}
+				if event.Has(fsnotify.Remove) {
+					//wait for file recreation
+					//REMOVE are used in k8s environment by configmap
+					for {
+						log.Infof("Waiting for file to be created: %s", filename)
+						time.Sleep(2 * time.Second)
+						_, err := os.Stat(filename)
+						if err == nil {
+							log.Infof("file created: %s", filename)
+							defer StartWatcher(filename, env)
+							close(quit)
+							break
+						}
+					}
+				} else if !event.Has(fsnotify.Write) {
+					//discard non-Write events
+					continue
+				}
+				log.Infof("Reloading config file: %s", filename)
+
+				yamlFile, err := os.ReadFile(filename)
+				if err != nil {
+					log.Errorf("err: %s", err)
+					return
+				}
+
+				err = yaml.Unmarshal(yamlFile, &envs)
+				if err != nil {
+					log.Errorf("err: %s", err)
+					return
+				}
+
+				log.Infoln("Load new SQS config:")
+				err = createSqsQueues(env)
```

Comment:
Also, I had a thought today - do we want to wipe the old Queues and Topics before reload the new data into them?  I figure, if something was removed, we probably want to yank that on a hot reload?  Maybe it's as easy as overwriting the queues with an empty map before repopulating?

Reply:
Further testing shows that it does this already.